### DISCUSSION TITLE
Explicitly Set-Cookie ("remember" behavior)

### DISCUSSION THREAD

**Discussion by dvnrsn** [sergiodxa/remix-auth#255] [Q&A] [answered]
In the default Indie Stack I notice this [createUserSession](https://github.com/remix-run/indie-stack/blob/main/app/session.server.ts#L66C3-L66C3) which includes a mechanism for setting cookie expiration (max-age)

```ts
const session = await getSession(request);
  session.set(USER_SESSION_KEY, userId);
  return redirect(redirectTo, {
    headers: {
      "Set-Cookie": await sessionStorage.commitSession(session, {
        maxAge: remember
          ? 60 * 60 * 24 * 7 // 7 days
          : undefined,
      }),
    },
  });
```

I am trying to figure out how to implement the same in remix-auth. Would it be a matter of changing headers in the action of login? Is this [advanced usage example](https://github.com/sergiodxa/remix-auth#custom-redirect-url-based-on-the-user) gleaning something useful?

**Comment by sergiodxa** [comment]
You need to do something similar to the advanced usage examples, don't use successRedirect and instead manually save the data to the session.

**Reply by dvnrsn** [reply, marked as answer]
Ok this works (edited because I had forgotten the very important `session.set(authenticator.sessionKey, user);`

```ts
const user = await authenticator.authenticate("user-pass", request, {
    failureRedirect: "/join",
    context: { formData },
  });

const session = await sessionStorage.getSession(
  request.headers.get("Cookie"),
);
session.set(authenticator.sessionKey, user);
const headers = new Headers({
  "Set-Cookie": await sessionStorage.commitSession(session, {
    maxAge: remember ? 30 * 24 * 60 * 60 : undefined,
  }),
});
return redirect(redirectTo, { headers });
```
<img width="125" alt="image" src="https://github.com/sergiodxa/remix-auth/assets/9112811/01c9a03f-ee10-46dc-828d-2a285ef140b0">
